src/Resizer.cs

```diff
@@ -1,27 +1,56 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 using System.IO;
-using ImageSharp;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Formats;
+using SixLabors.ImageSharp.Formats.Bmp;
+using SixLabors.ImageSharp.Formats.Gif;
+using SixLabors.ImageSharp.Formats.Jpeg;
+using SixLabors.ImageSharp.Formats.Png;
+using SixLabors.Primitives;
 
 namespace ImageResizer
 {
-    public static class Resizer
+    public class Resizer
     {
-        public static void Resize(string filePath, IEnumerable<int> widths)
+        private Image<Rgba32> _image;
+        private IImageEncoder _encoder;
+
+        public void Resize(string filePath, IEnumerable<int> widths)
         {
             using (var stream = File.OpenRead(filePath))
             {
-                var image = Image.Load(stream);
-                var aspectRatio = new Size(image.Width, image.Height).AspectRatio();
+                _image = Image.Load(stream);
+
+                var filetype = Path.GetExtension(filePath);
+
+                switch (filetype)
```

Comment:
can you break this `switch` block out into a "GetEncoder" method, called from the constructor?